course05/homework01/freekpaans/string.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <string>
+#include <ostream>
+#include <istream>
+
+class String
+{
+public:
+	String();
+	String(std::string);
+
+	bool Empty() const;
+	std::size_t Size() const;
+
+	bool operator==(const String) const; // These two copy the comparand, but we didn't cover references yet, right :)
+	bool operator!=(const String) const;
+
+	String Substring(int start) const;
+	String Substring(int start, int length) const;
```

Comment:
If your `size()` returns `size_t`, you should also use `size_t` here - otherwise you might run into issues when different size or signedness. We didnt cover this yet but every STL container has a type alias `size_type` that you would in general use in the class